Client/Application/Delegates/AppDelegate.swift

```diff
@@ -478,6 +478,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
         if let authInfo = KeychainWrapper.sharedAppContainerKeychain.authenticationInfo(), authInfo.isPasscodeRequiredImmediately {
             authenticator?.willEnterForeground()
         }
+        
+        AdblockResourceDownloader.shared.startLoading()
```

Comment:
I have a suggestion. what do you think about adding resource dowloading into `applicationDidEnterBackground`.

Use background task and use that window efficiently. We can define a background task with 

    private var backgroundTaskIdentifier: UIBackgroundTaskIdentifier?
    backgroundTaskIdentifier = UIApplication.shared.beginBackgroundTask {
        self.endBackgroundTask()
     }

and have en background task method like

    private func endBackgroundTask() {
        guard let backgroundTaskIdentifier = backgroundTaskIdentifier else { return }
        UIApplication.shared.endBackgroundTask(backgroundTaskIdentifier)
        self.backgroundTaskIdentifier = nil
    }

and also we can have completion block from AdBlockResourceDownloader and end the task If operation is success.

And since we have also resource downloading in foreground this will help us improving the download experience and utilize the background window

Reply:
overkill imo, why would you want to download it both when the app goes in background and in foreground?

Reply:
If we will download while going to background using background task, it will not need to download while app is coming foreground and occupy a thread and do processing. 
Why we need to add both(background/foreground) is background task can initiate processes that doesnt take too long and there is no guarantee it will succeed so in that case foreground can initialize the process if it is needed.

But yes this is a suggestion, not necessary.